src/classes/ACCT_Accounts_TDTM.cls

```diff
@@ -61,12 +61,38 @@ public class ACCT_Accounts_TDTM extends TDTM_Runnable {
         configAccountForContact(listNew, listOld, triggerAction, listHHNew, listHHOld);
 
         // deal with any name changes for Household Accounts
-        if (listHHNew.size() > 0)
+        if (listHHNew.size() > 0) {
             HH_HouseholdNaming.handleHHNameUpdates(listHHNew, listHHOld, triggerAction, objResult);
+        }
+
+        updateRecurringDonationsName(listNew, listOld, triggerAction);
 
         return null;
     }
 
+    /***
+     * @description Update Recurring Donations Name on Account name change.
+     * Applicable Accounts are not households.
+     */
+    private void updateRecurringDonationsName(List<SObject> accounts, List<SObject> oldAccounts, TDTM_Runnable.Action triggerAction) {
+        if (triggerAction != TDTM_Runnable.Action.AfterUpdate) {
+            return;
+        }
+
+        Map<Id, SObject> oldAccountById = new Map<Id, SObject>(oldAccounts);
+        List<Id> accountIds = new List<Id>();
+
+        for (Account acc : (List<Account>) accounts) {
+            if (acc.npe01__SYSTEM_AccountType__c != CAO_Constants.HH_ACCOUNT_TYPE
+                && acc.Name != oldAccountById.get(acc.Id).get('Name')
+            ) {
+                accountIds.add(acc.Id);
+            }
+        }
+
+        new RD2_NamingService().updateOnAccountChange(accountIds);
```

Comment:
There is a potential - though small - for this synchronous action to cause issues with governor limits such as SOQL, DML or CPU timeout as well as just general performance. I recommend implementing support for an asynchronous method in the naming service and use that for all naming changes that originate from the Account or Contact object.

Reply:
Updating RD names in a queueable is causing many HH and Contact test classes to fail due to "Max stack depth  has been reached"  and "Too many queueable jobs have been added to the queue: 2".
Also, when a HH naming batch is run, it seems better to update RD name instead creating queueable job for every `execute()` in the batch.